packages/next/client/dev/fouc.ts

```diff
@@ -1,12 +1,24 @@
-// This wrapper function is used to avoid raising a Trusted Types violation.
-const safeSetTimeout = (callback: () => void) => setTimeout(callback)
+// This wrapper function is used to safely select the best available function
```

Comment:
Summary of the issue:
1. Hidden iframes are not guaranteed to ever run callbacks schedule through `requestAnimationFrame`. [This table](https://www.vector-logic.com/blog/posts/on-request-animation-frame-and-embedded-iframes#browser_variations) breaks it down browser to browser. For example, on Chrome, `window.requestAnimationFrame` is available, but the callback is never called in a hidden cross-origin iframe.
2. `displayContent` (below) is passed as an option to `beforeRender` which is [specifically `await`-ed](https://github.com/vercel/next.js/blob/0f99768f9811fdf027a083471bd2a435f6207f4a/packages/next/client/index.tsx#L436) before hydrating the app. If `displayContent` never resolves, the app **never** hydrates.